### ISSUE TITLE
Generic File Maker 

### ISSUE THREAD

**Issue by CryptoPunkFrens** [maximedrn/opensea-automatic-bulk-upload-and-sale#356]
I have another problem Generic File Maker if you can help.
Choose your data file:
0 - Browse a file on PC.
File number: 0
Browsing on your computer...
Something went wrong. Expecting ',' delimiter: line 43 column 22 (char 1064)

**Comment by maximedrn** [comment]
Check in your generic JSON file at line 43, you forgot a coma at end of the line

**Comment by CryptoPunkFrens** [comment]
which JSON file? is this - generic-file-blank

**Comment by maximedrn** [comment]
Yes, the JSON file you completed

**Comment by CryptoPunkFrens** [comment]
I have coma on that file

**Comment by CryptoPunkFrens** [comment]
{
    "details": {
      "file_path": {
        "file_path": "C:/Users/Design3/Desktop/NFT/Black Punks/hashlips_art_engine_ps_script-1.0.1/build/images/{}.png",
        "incrementation": "number",
        "format": 0
      },
      "preview": {
        "preview": "",
        "incrementation": "",
        "format": 0
      },
      "nft_name": {
        "nft_name": "BlackPunks #{}",
        "incrementation": "number",
        "format": 0
      },
      "external_link": {
        "external_link": "",
        "incrementation": "",
        "format": 0
      },
      "description": {
        "description": "",
        "incrementation": "",
        "format": 0
      },
      "collection": "",
      "properties": [],
      "levels": [],
      "stats": [],
      "unlockable_content": {
        "unlockable_content": "",
        "incrementation": "",
        "format": 0
      },
      "explicit_and_sensitive_content": false,
      "supply": 1,
      "blockchain": "Polygon",
      "sale_type": [],
      "price": 0.01,
      "method": [],
      "duration": [6 months],
      "specific_buyer": "",
      "quantity": 1
    },
    "settings": {
        "overwrite": false,
        "overwrite_empty": true,
        "append": false
    }
}

**Comment by maximedrn** [comment]
`"duration": [6 months],` has to be `"duration": ["6 months"],` or `"duration": "6 months",`

**Comment by maximedrn** [comment]
What metadata file did you use?

**Comment by CryptoPunkFrens** [comment]
didn't ask me to choose
I follow your video and I did see that step is missing
What I did wrong?

**Comment by maximedrn** [comment]
It asks you to choose a metadata file

**Comment by CryptoPunkFrens** [comment]
ok let me try again

**Comment by CryptoPunkFrens** [comment]
which account number is this?
What is your account number? (Press [ENTER] to ignore this step)
I ignored last time

**Comment by maximedrn** [comment]
If you want to switch from account 1 to any other account, you can enter a value. Otherwise, ignore it.
The metadata file is the second thing asked.

**Comment by CryptoPunkFrens** [comment]
I did choose metadata already
but what is that account number?? is that metamask wallet or something else?

**Comment by maximedrn** [comment]
It corresponds to the name of the accounts linked to your MetaMask wallet (Account 1, 2, 3, etc).
The first time you use the bot, there is only Account 1. If you want to use another account, you have to import it using its private key.

**Comment by CryptoPunkFrens** [comment]
so I definitely need that in the future, because I will be using more accounts

**Comment by CryptoPunkFrens** [comment]
and maybe there was the problem because I did see logged on opensee with main account or account 1 and I don't want to use that account

**Comment by CryptoPunkFrens** [comment]
now I have this
Login to MetaMask. Login to MetaMask failed. Retrying.
Login to MetaMask. Login to MetaMask failed. Restarting.
Something went wrong with your webdriver.
Message: unknown error: unable to discover open window in chrome
  (Session info: chrome=103.0.5060.114)
Stacktrace:
Backtrace:
        Ordinal0 [0x00EE6463+2188387]
        Ordinal0 [0x00E7E461+1762401]
        Ordinal0 [0x00D93D78+802168]
        Ordinal0 [0x00D7BF35+704309]
        Ordinal0 [0x00DE16C8+1119944]
        Ordinal0 [0x00DE108A+1118346]
        Ordinal0 [0x00DDC5F6+1099254]
        Ordinal0 [0x00DB6BE0+945120]
        Ordinal0 [0x00DB7AD6+948950]
        GetHandleVerifier [0x011871F2+2712546]
        GetHandleVerifier [0x0117886D+2652765]
        GetHandleVerifier [0x00F7002A+520730]
        GetHandleVerifier [0x00F6EE06+516086]
        Ordinal0 [0x00E8468B+1787531]
        Ordinal0 [0x00E88E88+1805960]
        Ordinal0 [0x00E88F75+1806197]
        Ordinal0 [0x00E91DF1+1842673]
        BaseThreadInitThunk [0x7586FA29+25]
        RtlGetAppContainerNamedObjectPath [0x77657A9E+286]
        RtlGetAppContainerNamedObjectPath [0x77657A6E+238]

**Comment by maximedrn** [comment]
Send me the metadata file used

**Comment by CryptoPunkFrens** [comment]
how to send you, don't support file type

**Comment by maximedrn** [comment]
ZIP it in an archive

**Comment by maximedrn** [comment]
Yes, but send me the file in a ZIP

**Comment by maximedrn** [comment]
Right click, zip in archive. Or using WinRAR, add it in a ZIP archive, not a RAR

**Comment by CryptoPunkFrens** [comment]
[upload_and_sale_4b3b79ef.zip](https://github.com/maximedrn/opensea-automatic-bulk-upload-and-sale/files/9070820/upload_and_sale_4b3b79ef.zip)

**Comment by maximedrn** [comment]
It is an empty file:

```json
{
    "nft": [
        {
            "file_path": "",
            "nft_name": "",
            "external_link": "",
            "description": "",
            "collection": "",
            "properties": [],
            "levels": [],
            "stats": [],
            "unlockable_content": [
                false,
                ""
            ],
            "explicit_and_sensitive_content": false,
            "supply": 0,
            "blockchain": "",
            "sale_type": "",
            "price": 0,
            "method": [],
            "duration": [],
            "specific_buyer": "",
            "quantity": 0
        }
    ]
}
```

**Comment by maximedrn** [comment]
I think you selected the wrong generic.json file when generating your metadata file.

**Comment by CryptoPunkFrens** [comment]
I also have this
[upload_and_sale_a889e8ce.zip](https://github.com/maximedrn/opensea-automatic-bulk-upload-and-sale/files/9070857/upload_and_sale_a889e8ce.zip)

**Comment by maximedrn** [comment]
Same, look at the `data/` folder of the Generic File Maker folder, the correct metadata file should be in

**Comment by CryptoPunkFrens** [comment]
[5286840b.zip](https://github.com/maximedrn/opensea-automatic-bulk-upload-and-sale/files/9070877/5286840b.zip)

**Comment by CryptoPunkFrens** [comment]
this is full data but cmd is not giving me option to select that file

**Comment by maximedrn** [comment]
This one is perfect. When selecting your file choose the 0 option and browse the file on your computer

**Comment by maximedrn** [comment]
Moreover make sure that you do not want to set a collection because this field is empty

**Comment by CryptoPunkFrens** [comment]
I don't understand 
"Moreover make sure you do not want to have collection because this field is empty"

**Comment by maximedrn** [comment]
The collection field in your metadata file is empty so OpenSea will create an Untitled collection for all of your NFTs.

**Comment by CryptoPunkFrens** [comment]
oh ok so I need to put collection name
you didn't do it in your video :-)

**Comment by maximedrn** [comment]
You can complete the other field. For the collection it works like this:
`https://opensea.io/collection/this-is-my-colllection` -> `"collection": "this-is-my-collection",`

**Comment by CryptoPunkFrens** [comment]
so now I need to create again, is that what you telling me

**Comment by maximedrn** [comment]
> so now I need to create again, is that what you telling me

Complete the collection field in the generic.json file and run the generic.py file again to generate the new metadata file.

**Comment by CryptoPunkFrens** [comment]
is it correct?
[generic-file-blank.zip](https://github.com/maximedrn/opensea-automatic-bulk-upload-and-sale/files/9070994/generic-file-blank.zip)

**Comment by CryptoPunkFrens** [comment]
please confirm before I run new metadata

**Comment by maximedrn** [comment]
Correct 👌

**Comment by CryptoPunkFrens** [comment]
ok let me do it

**Comment by CryptoPunkFrens** [comment]
Created new metadata. Please can you check
[9bb5609f.zip](https://github.com/maximedrn/opensea-automatic-bulk-upload-and-sale/files/9071247/9bb5609f.zip)

**Comment by maximedrn** [comment]
> Created new metadata. Please can you check [9bb5609f.zip](https://github.com/maximedrn/opensea-automatic-bulk-upload-and-sale/files/9071247/9bb5609f.zip)

Correct too

**Comment by CryptoPunkFrens** [comment]
is uploading but I don't know where because the https://opensea.io/collection/blackpunksnft is empty

**Comment by CryptoPunkFrens** [comment]
NFT n°1/5000:
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Metadata saved in the data/upload_and_sale_bec0b2c3.json file.

NFT n°2/5000:
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Metadata saved in the data/upload_and_sale_bec0b2c3.json file.

NFT n°3/5000:
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Metadata saved in the data/upload_and_sale_bec0b2c3.json file.

NFT n°4/5000:
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Metadata saved in the data/upload_and_sale_bec0b2c3.json file.

NFT n°5/5000:
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Metadata saved in the data/upload_and_sale_bec0b2c3.json file.

NFT n°6/5000:
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Metadata saved in the data/upload_and_sale_bec0b2c3.json file.

NFT n°7/5000:
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Uploading NFT. Solving the reCAPTCHA. Solved. Upload failed.
Metadata saved in the data/upload_and_sale_bec0b2c3.json file.

**Comment by maximedrn** [comment]
What error is displayed by OpenSea?

**Comment by CryptoPunkFrens** [comment]
All what I can see is this because is not going to collection page
This error is on the first page just before Create ; [401] Not authorized contract editor.

**Comment by CryptoPunkFrens** [comment]
but metamask is on main account is not on account for this collection

**Comment by CryptoPunkFrens** [comment]
how to stop this process?

**Comment by maximedrn** [comment]
> how to stop this process?

Ctrl + C

**Comment by maximedrn** [comment]
> but metamask is on main account is not on account for this collection

What information did you type for the account name? Its name or its private key (not the recovery phrase)?

**Comment by maximedrn** [comment]
So you want to upload on your Account 1, right?

**Comment by CryptoPunkFrens** [comment]
anything will do, what is better way. I can even create collection on main account and just to run again

**Comment by maximedrn** [comment]
Yes, you can do this. Make sure to update the collection field in the metadata file if you change the collection URL

**Comment by CryptoPunkFrens** [comment]
will be the same name because I will close first this collection on this account and then I will create the same name of the collection on my main Account1

**Comment by CryptoPunkFrens** [comment]
I will prefer if I can keep this account3 for this collection, so what I need to change in cmd

**Comment by maximedrn** [comment]
Open the MetaMask extension (not in the bot), click on the 3 dots, select Account details, and click on Export private key.
Copy it and paste it when it asks for the account name.

**Comment by CryptoPunkFrens** [comment]
thanks a lot
It's running nicely
Some of them is failing but that is the problem for another day. I have collection of 5000 so I will contact you again to help me with this images to upload them. You are very helpful and I will give 5 stars everywhere, for now ciao and don't forget to enjoy your life ;-)

**Comment by CryptoPunkFrens** [comment]
hi please help if you there
My PC shut down so it's stop uploading
How to care on from where stopped?

**Comment by CryptoPunkFrens** [comment]
Hi
Today on another PC I have this problem:
NFT n°1/20000:
Uploading NFT. Upload failed. Collection doesn't exist or can't be found.
Uploading NFT. Upload failed. Collection doesn't exist or can't be found.
Metadata saved in the data/upload_and_sale_cd1ca208.json file.

NFT n°2/20000:
Uploading NFT. Upload failed. Collection doesn't exist or can't be found.
Uploading NFT. Upload failed. Collection doesn't exist or can't be found.
Metadata saved in the data/upload_and_sale_cd1ca208.json file.

NFT n°3/20000:
Uploading NFT. Upload failed. Collection doesn't exist or can't be found.
Uploading NFT. Upload failed. Collection doesn't exist or can't be found.
Metadata saved in the data/upload_and_sale_cd1ca208.json file.

NFT n°4/20000:
Uploading NFT. Upload failed. Collection doesn't exist or can't be found.
Uploading NFT. Upload failed. Collection doesn't exist or can't be found.
Metadata saved in the data/upload_and_sale_cd1ca208.json file.

**Comment by CryptoPunkFrens** [comment]
I have to ask can I upload another collection on another PC with different metamask account???

**Comment by CryptoPunkFrens** [comment]
then I did something wrong
I have error on approving collection, my metamask comes on but is not approving

**Comment by maximedrn** [comment]
You must approve your collection manually. OpenSea asks for it once.

**Comment by CryptoPunkFrens** [comment]
even when I try to do that is the same error

**Comment by CryptoPunkFrens** [comment]
why I can't approve this collection? I trayed on all 3 PC and is just not possible

**Comment by CryptoPunkFrens** [comment]
what is the problem? is it wallet or setting up collection??

**Comment by maximedrn** [comment]
Do you get a red message when you try to approve your collection?

**Comment by CryptoPunkFrens** [comment]
no wasn't red was black pop-up on the right bottom corner to try again latter
I change account and worked, so maybe something is wrong with that account

**Comment by CryptoPunkFrens** [comment]
I have one BIG question for you
How to generate the metadata for nft without layers. All what I have is already made NFT and I have the name of that NFT. I did 500 Celebrity's in 30 different effects in photoshop, I want to have metadata on opensea on the left hand side buyer can choose which effect they want to look and then only 500 of celebrities can show. Please any idea how to do that. I'm trying already to do this for more then a month and I just can't do it. Please help if you can

**Comment by maximedrn** [comment]
Properties can be added manually but it takes time.. That's why I use HashLips.
Otherwise, you can follow this structure for the properties: 
```json
{
    "nft": [
        ...
        "properties": [
             { "type": "Dog", "name": "Male" },
             { "type": "Cat", "name": "Female" },
             ...
             { "type": "Dog", "name": "Male" }
         ],
         ...
    ]
}
```

**Comment by CryptoPunkFrens** [comment]
I tried in HashLips by creating layers with nothing on it like: layer: name of the effect, layer: the year, layer: male or female, layer: background and layer: singer, actor and sport celebrity's. But didn't come right, I don't know where I went wrong Maybe you can point how to do this way properly

**Comment by maximedrn** [comment]
Use transparent layers. For example for "Year", create a folder named Year and place different transparent `.png` images with different names in it.

**Comment by CryptoPunkFrens** [comment]
Do I need to buy Collection Scraper to find missing/duplicated nft's and then I need to buy Deletion to use that file what Scraper created?

**Comment by CryptoPunkFrens** [comment]
I have problem running Screper
This is the error
C:\Users\Design3\Desktop\Uploading to Opensea\Collection_Scraper\opensea-collection-scraper>python scraper.py
python: can't open file 'C:\\Users\\Design3\\Desktop\\Uploading to Opensea\\Collection_Scraper\\opensea-collection-scraper\\scraper.py': [Errno 2] No such file or directory

C:\Users\Design3\Desktop\Uploading to Opensea\Collection_Scraper\opensea-collection-scraper>node index.js --collection any_collection_you_want
internal/modules/cjs/loader.js:905
  throw err;
  ^

Error: Cannot find module './opensea'
Require stack:
- C:\Users\Design3\Desktop\Uploading to Opensea\Collection_Scraper\opensea-collection-scraper\index.js
←[90m    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)←[39m
←[90m    at Function.Module._load (internal/modules/cjs/loader.js:746:27)←[39m
←[90m    at Module.require (internal/modules/cjs/loader.js:974:19)←[39m
←[90m    at require (internal/modules/cjs/helpers.js:93:18)←[39m
    at Object.<anonymous> (C:\Users\Design3\Desktop\Uploading to Opensea\Collection_Scraper\opensea-collection-scraper\index.js:2:17)
←[90m    at Module._compile (internal/modules/cjs/loader.js:1085:14)←[39m
←[90m    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)←[39m
←[90m    at Module.load (internal/modules/cjs/loader.js:950:32)←[39m
←[90m    at Function.Module._load (internal/modules/cjs/loader.js:790:12)←[39m
←[90m    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)←[39m {
  code: ←[32m'MODULE_NOT_FOUND'←[39m,
  requireStack: [
    ←[32m'C:\\Users\\Design3\\Desktop\\Uploading to Opensea\\Collection_Scraper\\opensea-collection-scraper\\index.js'←[39m
  ]
}

C:\Users\Design3\Desktop\Uploading to Opensea\Collection_Scraper\opensea-collection-scraper>

**Comment by maximedrn** [comment]
Files are not in the same directory https://github.com/maximedrn/opensea-automatic-bulk-upload-and-sale/discussions/340

**Comment by CryptoPunkFrens** [comment]
thanks, yes was 3 files in another folder

**Comment by CryptoPunkFrens** [comment]
so now I have my files from Scraper, and I placed Delete in to processes Folder
How I'm running delete faction????

**Comment by maximedrn** [comment]
Run the main.py, select the one of the duplicate files and select the 4th option (delete/remove NFT)

**Comment by CryptoPunkFrens** [comment]
it's running
THIS IS ALL VERY FANTASTIC
THANK YOU

**Comment by CryptoPunkFrens** [comment]
what mines when say " NFT cannot be deleted"

**Comment by CryptoPunkFrens** [comment]
oh another question
If I have collection without metadata what I need to do

**Comment by Bogdandp** [comment]
hello, have a problem: 
NFT n°2/2:
Your file is poorly structured for this NFT.
Check that elements are present and in the right order.

my meta: 
  {
      "file_path": "D:/Users10/User/Desktop/bot/opensea-automatic-bulk-upload-and-sale/meta.png",
      "nft_name": "NFT #1",
      "description": "This is my first NFT.",
      "collection": "My First NFT",
      "supply": 1,
      "blockchain": "Polygon"
    },

**Comment by maximedrn** [comment]
> oh another question If I have collection without metadata what I need to do

You're still able to create your metadata file.

**Comment by maximedrn** [comment]
> hello, have a problem: NFT n°2/2: Your file is poorly structured for this NFT. Check that elements are present and in the right order.
> 
> my meta: { "file_path": "D:/Users10/User/Desktop/bot/opensea-automatic-bulk-upload-and-sale/meta.png", "nft_name": "NFT #1", "description": "This is my first NFT.", "collection": "My First NFT", "supply": 1, "blockchain": "Polygon" },

It should have this format: https://github.com/maximedrn/opensea-automatic-bulk-upload-and-sale/blob/master/data/examples/json_structure_upload_and_sale.json with **all* the details, even empty.

**Comment by CryptoPunkFrens** [comment]
> > oh another question If I have collection without metadata what I need to do
> 
> You're still able to create your metadata file.

but I don't have json file, how to do it?

**Comment by maximedrn** [comment]
You can create a metadata file from scratch using the Generic File Maker

**Comment by CryptoPunkFrens** [comment]
but I don't want metadata for this collection, I just don't know how to upload the images nft's without metadata

**Comment by CryptoPunkFrens** [comment]
or if I need to make metadata to be able to upload, please show me step by step
